### PR TITLE
api: Add some missing struct member documentation

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: lib/fizzy/instantiate.hpp [hunk 1]
```diff
@@ -81,20 +81,28 @@ using bytes_ptr = std::unique_ptr<bytes, void (*)(bytes*)>;
...
 {
+    /// Module of this instance.
     std::unique_ptr<const Module> module;
-    // Memory is either allocated and owned by the instance or imported as already allocated bytes
-    // and owned externally.
-    // For these cases unique_ptr would either have a normal deleter or noop deleter respectively
+    /// Instance memory.
+    /// Memory is either allocated and owned by the instance or imported as already allocated bytes
+    /// and owned externally.
+    /// For these cases unique_ptr would either have a normal deleter or noop deleter respectively
```
Comment (axic) [inline lib/fizzy/instantiate.hpp:89]:
```suggestion
    /// For these cases unique_ptr would either have a normal deleter or no-op deleter respectively
```

Comment (axic) [reply lib/fizzy/instantiate.hpp:89]: Is it "deleter" or destructor?

File: lib/fizzy/instantiate.hpp [hunk 2]
```diff
@@ -81,20 +81,28 @@ using bytes_ptr = std::unique_ptr<bytes, void (*)(bytes*)>;
...
     bytes_ptr memory = {nullptr, [](bytes*) {}};
+    /// Memory limits.
     Limits memory_limits;
-    // Hard limit for memory growth in pages, checked when memory is defined as unbounded in module
+    /// Hard limit for memory growth in pages, checked when memory is defined as unbounded in module
```
Comment (axic) [inline lib/fizzy/instantiate.hpp:93]:
```suggestion
    /// Hard limit for memory growth in pages, checked when memory is defined as unbounded in module.
```

File: lib/fizzy/instantiate.hpp [hunk 3]
```diff
@@ -81,20 +81,28 @@ using bytes_ptr = std::unique_ptr<bytes, void (*)(bytes*)>;
 /// The module instance.
 struct Instance
 {
+    /// Module of this instance.
     std::unique_ptr<const Module> module;
-    // Memory is either allocated and owned by the instance or imported as already allocated bytes
-    // and owned externally.
-    // For these cases unique_ptr would either have a normal deleter or noop deleter respectively
+    /// Instance memory.
```
Comment (chfast) [inline lib/fizzy/instantiate.hpp:86]: I'd insist on separating these by an empty line.

File: lib/fizzy/instantiate.hpp [hunk 4]
```diff
@@ -81,20 +81,28 @@ using bytes_ptr = std::unique_ptr<bytes, void (*)(bytes*)>;
...
+    /// Memory limits.
     Limits memory_limits;
-    // Hard limit for memory growth in pages, checked when memory is defined as unbounded in module
+    /// Hard limit for memory growth in pages, checked when memory is defined as unbounded in module
     uint32_t memory_pages_limit = 0;
-    // Table is either allocated and owned by the instance or imported and owned externally.
-    // For these cases unique_ptr would either have a normal deleter or noop deleter respectively.
+    /// Instance table.
+    /// Table is either allocated and owned by the instance or imported and owned externally.
+    /// For these cases unique_ptr would either have a normal deleter or noop deleter respectively.
```
Comment (axic) [inline lib/fizzy/instantiate.hpp:97]:
```suggestion
    /// For these cases unique_ptr would either have a normal deleter or no-op deleter respectively.
```

File: lib/fizzy/instantiate.hpp [hunk 5]
```diff
@@ -81,20 +81,28 @@ using bytes_ptr = std::unique_ptr<bytes, void (*)(bytes*)>;
...
     table_ptr table = {nullptr, [](table_elements*) {}};
+    /// Table limits.
     Limits table_limits;
+    /// Instance globals (doesn't include imported globals).
```
Comment (axic) [inline lib/fizzy/instantiate.hpp:101]:
```suggestion
    /// Instance globals (excluding imported globals).
```

File: lib/fizzy/instantiate.hpp [hunk 6]
```diff
@@ -124,10 +132,15 @@ std::unique_ptr<Instance> instantiate(std::unique_ptr<const Module> module,
 /// Function that should be used by instantiate as import, identified by module and function name.
 struct ImportedFunction
 {
+    /// Module name.
     std::string module;
+    /// Function name.
     std::string name;
+    /// Input types array.
```
Comment (axic) [inline lib/fizzy/instantiate.hpp:139]:
```suggestion
    /// Array of input types.
```

File: lib/fizzy/instantiate.hpp [hunk 7]
```diff
@@ -124,10 +132,15 @@ std::unique_ptr<Instance> instantiate(std::unique_ptr<const Module> module,
...
     std::vector<ValType> inputs;
+    /// Output type or empty optional if function has void return type.
     std::optional<ValType> output;
+    /// Function object, which will be called for executing a function.
```
Comment (axic) [inline lib/fizzy/instantiate.hpp:143]:
```suggestion
    /// Function object, which will be called to execute the function.
```

File: lib/fizzy/instantiate.hpp (hunk 1 above)
Comment (gumb0) [reply lib/fizzy/instantiate.hpp]: > Is it "deleter" or destructor?

It refers to `unique_ptr`'s deleter

image

**Review by axic — APPROVED** [review]

**Comment by gumb0** [comment]
Added empty lines.

**Review by axic — APPROVED** [review]

**Pull request merged**